tapscript/tx.go

```diff
@@ -263,8 +308,19 @@ func InputAssetPrevOut(prevAsset asset.Asset) (*wire.TxOut, error) {
 	var pkScript []byte
 	switch prevAsset.ScriptVersion {
 	case asset.ScriptV0:
-		var err error
-		pkScript, err = PayToTaprootScript(prevAsset.ScriptKey.PubKey)
+		var (
+			err           error
+			validationKey = prevAsset.ScriptKey.PubKey
+		)
+
+		// If the input asset is a genesis asset that is part of an
+		// asset group, we need to validate the group witness against
+		// the tweaked group key and not the genesis asset script key.
+		if prevAsset.HasGenesisWitnessForGroup() {
```

Comment:
My idea here was to set the `prevAsset.ScriptKey = groupKey` when creating the `prevAsset` copy. Is that feasible?

Reply:
Covered here:

https://github.com/Roasbeef/bips/pull/33#discussion_r1259002226

I think this still needs to get reflected in the BIP text. AFAICT, the `pkScript` here is what we actually pass to the internal `txscript.Engine` to validate the witness against, so it's the one we need to now be the group key. The asset used to build the virtual TX should have the tweaked group key set, but not swapped for the script key.

Reply:
See my comment linked on the BIP:

> The modification only affects the referenced previous leaf, that is used to construct the previous outpoint for the virtual tx. The output leaf (still committing to asset_script_key) remains the same.

Reply:
Resolved in BIP repo.